src/main/java/com/onkibot/backend/database/entities/ExternalResource.java

```diff
@@ -26,6 +27,10 @@
   @JoinColumn(name = "publisher_user_id")
   private User publisherUser;
 
+  @OneToMany(mappedBy = "externalResourceApprovalId.externalResource", fetch = FetchType.LAZY)
```

Comment:
The OneToMany should include a CASCADE. REMOVE so that the approval gets removed when an external resource is deleted (if we still support removal of external resources).